server/resources/resource_management.go

```diff
@@ -593,30 +593,30 @@ func getFilterFromPolicy(
 	return filter
 }
 
-func checkIfResourceExistsForPolicy(
+func fetchResourceIfExistsWithPolicy(
 	context context.Context,
 	auth schema.Authorization,
 	resourceID string,
 	resourceSchema *schema.Schema,
 	policy *schema.Policy,
 	action string,
 	preTransaction transaction.Transaction,
-) (bool, error) {
+) (*schema.Resource, error) {
 	filter := getFilterFromPolicy(auth, resourceID, resourceSchema, policy, action)
 
-	return checkIfResourceExists(context, filter, resourceSchema, preTransaction)
+	return fetchResourceIfExists(context, filter, resourceSchema, preTransaction)
 }
 
-func checkIfResourceExists(context context.Context, filter transaction.Filter,
-	resourceSchema *schema.Schema, preTransaction transaction.Transaction) (bool, error) {
-	_, err := preTransaction.Fetch(context, resourceSchema, filter, nil)
+func fetchResourceIfExists(context context.Context, filter transaction.Filter,
```

Comment:
There is no need for this change. AFAICT this was added to have resource contents in `UpdateResource` in order to be able to create an instance of `Tenancy`. But resource is already fetched inside `UpdateResourceInTransaction` so `Tenancy` can actually be created there.

Reply:
Right, I've overlook that resource was fetched again, I've simplified it now.